### ISSUE TITLE
API class does not support array parameters

### ISSUE THREAD

**Issue by FarooqAlaulddin** [plone/volto#4349]
I am trying to use Volto API class to make a request to plone 6 endpoint where the payload is object that contain an array as one of properties. The API class does not seem to support that. Am I missing something?

https://github.com/plone/volto/blob/0f84668688ce1d69030d2769d32a01d7dde28964/src/helpers/Api/Api.js#L60

**Comment by tiberiuichim** [comment]
@FarooqAlaulddin it's not clear what the problem is. Can you add a bit more context? code sample, etc?

**Comment by FarooqAlaulddin** [comment]
@tiberiuichim sure! For example, I passed a similar params object to the API.get method:

     {
       searchTerm: 'power',
       page: 1,
       perPage: 10,
       fields: [
        {name: 'field1', value:'val for field 1'},
        {name: 'field2', value:'val for field 2'}
        {name: 'field3', value:'val for field 3'}
       ]
     }

This problem is the `fields` will not be treated correctly.
The Query String Prams will look something like this:
![image](https://user-images.githubusercontent.com/37892823/216383967-4b8e9bb4-88f9-45af-8550-3b817da9c2f3.png)

Instead of 
![image](https://user-images.githubusercontent.com/37892823/216384608-6b59842c-111d-4724-81a3-cb542eb2ec45.png)

**Comment by tiberiuichim** [comment]
it seems to be a problem with superagent.query. On top of that, Volto is using an old version of superagent, afaik, the changelog lists a change regarding this https://github.com/ladjs/superagent/releases/tag/v6.1.0

Would it be an option to serialize the query on your own?

**Comment by FarooqAlaulddin** [comment]
Yes! Serialize the queries before sending is the way to go. Thank you.

**Comment by tiberiuichim** [comment]
@FarooqAlaulddin would you please add a new "faq" entry in https://github.com/plone/volto/issues/2810 with this problem and the solution? Thank you!
